### ISSUE TITLE
Proxyconfig-controller fails to delete logcollector service account

### ISSUE THREAD

**Issue by ewolinetz** [openshift/cluster-logging-operator#375]
This may not be the core issue, however I see the following error in the cluster-logging-operator logs repeat several times during start up. It does eventually stop appearing, however maybe this is something we can handle more gracefully with a `get` check first?

```
{"level":"error","ts":1582132106.5039325,"logger":"kubebuilder.controller","msg":"Reconciler error","controller":"proxyconfig-controller","request":"/cluster","error":"Unable to create or update collection for \"\": Failure deleting logcollector service account: an empty namespace may not be set when a resource name is provided","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/github.com/go-logr/zapr/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:134\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/openshift/cluster-logging-operator/_output/src/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

**Comment by syedriko** [comment]
I've tracked this down to missing ClusterLogging instance, CLO uses it to find out the the ns of of the logcollector sa, which ends up missing. Working on a fix.

**Comment by syedriko** [comment]
> It does eventually stop appearing

This is the impression that exponential backoff can give when it kicks in for a repeatedly failing reconcile. Here's the sequence of timestamps on the log messages like above from one of my observations, with ```how-many x delay in seconds between retries``` added:

```
10x1
1584247163.3394256
1584247164.34169
1584247165.3430953
1584247166.3439848
1584247167.3464713
1584247168.3485808
1584247169.3515863
1584247170.3558254
1584247171.356483
1584247172.6368763
1x3
1584247175.1979172
1x5
1584247180.318368
1x10
1584247190.5595152
1x20
1584247211.0398045
1x40
1584247252.0001447
1x80
1584247333.9206667
1x160
1584247497.7611048
```

**Comment by syedriko** [comment]
@ewolinetz Could you take a look and may be close this?

**Comment by jcantrill** [comment]
fixed by #414
